lm_eval/models/indommlu_huggingface.py

```diff
@@ -0,0 +1,340 @@
+        trust_remote_code: Optional[bool] = False,
+        use_fast_tokenizer: Optional[bool] = True,
+        # arguments used for splitting a model across GPUs naively.
+        # only used if `parallelize=True`.
+        parallelize: Optional[bool] = False,
+        device_map_option: Optional[str] = "auto",
+        max_memory_per_gpu: Optional[Union[int, str]] = None,
+        max_cpu_memory: Optional[Union[int, str]] = None,
+        offload_folder: Optional[Union[str, os.PathLike]] = "./offload",
+        # PEFT and quantization options
+        peft: Optional[str] = None,
+        autogptq: Optional[Union[bool, str]] = False,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            pretrained=pretrained,
+            backend=backend,
+            revision=revision,
+            subfolder=subfolder,
+            tokenizer=tokenizer,
+            truncation=truncation,
+            max_length=max_length,
+            device=device,
+            dtype=dtype,
+            batch_size=batch_size,
+            max_batch_size=max_batch_size,
+            trust_remote_code=trust_remote_code,
+            use_fast_tokenizer=use_fast_tokenizer,
+            parallelize=parallelize,
+            device_map_option=device_map_option,
+            max_memory_per_gpu=max_memory_per_gpu,
+            max_cpu_memory=max_cpu_memory,
+            offload_folder=offload_folder,
+            peft=peft,
+            autogptq=autogptq,
+            **kwargs,
+        )
+
+    def _model_call(self, inps, attn_mask=None, labels=None):
+        """
+        :param inps: torch.Tensor
+            A torch tensor of shape [batch, (sequence_ctx + sequence_cont)] or of shape
+            [batch, sequence_ctx]. the size of sequence may vary from call to call
+        :param attn_mask: torch.Tensor, optional
+            A torch tensor of shape [batch, (sequence_ctx + sequence_cont)]. Only passed
+            (and must be passed) if self.AUTO_MODEL_CLASS is transformers.AutoModelForSeq2SeqLM
+        :param labels: torch.Tensor, optional
+            A torch tensor of shape [batch, (sequence_ctx + sequence_cont)]. Only passed
+            (and must be passed) if self.AUTO_MODEL_CLASS is transformers.AutoModelForSeq2SeqLM
+        :return
+            A torch tensor of shape [batch, sequence, vocab] with the
+        logits returned from the model's decoder
+        """
+        with torch.no_grad():
+            if attn_mask is not None or labels is not None:
+                assert attn_mask is not None and labels is not None
+                assert self.AUTO_MODEL_CLASS == transformers.AutoModelForSeq2SeqLM
+                return self.model(
+                    input_ids=inps, attention_mask=attn_mask, labels=labels
+                ).logits
+            else:
+                assert self.AUTO_MODEL_CLASS == transformers.AutoModelForCausalLM
+                outputs = self.model(inps).logits.to(torch.double)
+                return outputs
```

Comment:
Can consider doing the following to make that intention that we only modify the output for the CausalLM case, else we just use the parent's implementation/output.

```
outputs = super()._model_call(inps, attn_mask, labels)
if not (attn_mask is not None or labels is not None):
    outputs = outputs.to(torch.double)
return output
```